SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx

```diff
@@ -501,17 +499,24 @@ int vtkSlicerSkeletalRepresentationInitializerLogic::ShowFittingEllipsoid(vtkPoi
     }
     // compute best fitting ellipsoid
     // For now assume that the surface is centered and rotationally aligned
-    // 1. compute the second moment
-    Eigen::MatrixXd point_matrix_transposed = point_matrix.transpose();
-    Eigen::Matrix3d second_moment = point_matrix_transposed * point_matrix;
+    // 1. compute the second moment after centering the data points
+    Eigen::MatrixXd center = point_matrix.colwise().mean();
+    Eigen::MatrixXd centered_point_mat = point_matrix - center.replicate(point_matrix.rows(), 1);
+    Eigen::MatrixXd point_matrix_transposed = centered_point_mat.transpose();
+    Eigen::Matrix3d second_moment = point_matrix_transposed * centered_point_mat;
```

Comment:
I am not sure, after all the operations if the dimension of the matrix `second_moment` is known at compile time. If it is, I suggest changing the type of the matrix.